helm_deploy/prisoner-content-hub-backend/templates/db-refresh.yaml

```diff
@@ -0,0 +1,54 @@
+{{- if .Values.dbRefresh.enabled -}}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: db-refresh-script
+  labels:
+    {{- include "prisoner-content-hub-backend.labels" . | nindent 4 }}
+data:
+  entrypoint.sh: |-
+{{ .Files.Get "db-refresh.sh" | indent 4 }}
+---
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: {{ include "prisoner-content-hub-backend.fullname" . }}-db-refresh-cronjob
+  labels:
+    {{- include "prisoner-content-hub-backend.labels" . | nindent 4 }}
+spec:
+  schedule: "30 4 * * *"
+  concurrencyPolicy: Forbid
+  successfulJobsHistoryLimit: 5
+  failedJobsHistoryLimit: 5
+  jobTemplate:
+    spec:
+      # Tidy up all jobs after 4 days
+      ttlSecondsAfterFinished: 345600
+      backoffLimit: 0
+      activeDeadlineSeconds: 2400
+      template:
+        spec:
+          initContainers:
+          - name: db-refresh-cronjob
+            image: "ghcr.io/ministryofjustice/hmpps-mysql-tools:latest"
+            command:
+              - /bin/entrypoint.sh
+            volumeMounts:
+              - name: db-refresh-script
+                mountPath: /bin/entrypoint.sh
+                readOnly: true
+                subPath: entrypoint.sh
+  {{ include "db-backup.envs" . | nindent 12 }}
```

Comment:
The db refresh container is run as an `initContainer`.  This means it runs first, before the deployment.

The one thing to be aware of this method, is that it can be quite hard to obtain logs if something fails within the initContainer.  If you do `kubectl logs <pod-name>` you will just see that it failed to startup and no further messages.  To get the actual messages you need to do: `kubectl logs <pod-name> -c <init-container-2>`. See https://kubernetes.io/docs/tasks/debug/debug-application/debug-init-containers/